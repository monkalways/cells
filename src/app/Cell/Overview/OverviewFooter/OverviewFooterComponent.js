import React from 'react';
import PropTypes from 'prop-types';
import {
  AppBar, Button, Toolbar, withStyles, Grid,
} from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  onSignIn: PropTypes.func.isRequired,
};

const OverviewFooterComponent = ({ classes, isAuthenticated, onSignIn }) => (
  <div className={classes.root}>
    <AppBar position="static" className={classes.appBar}>
      <Toolbar>
        <Grid container>
          {isAuthenticated ? (
            <Grid item xs={4}>
              <Button
                variant="contained"
                color="secondary"
                className={classes.button}
              >
                You are authenticated
              </Button>
            </Grid>
          ) : (
            <Grid item xs={12}>
              <Button
                variant="contained"
                color="secondary"
                className={classes.button}
                onClick={() => onSignIn(true)}
              >
                Scan ID Card to Access
              </Button>
            </Grid>
          )}
        </Grid>
      </Toolbar>
    </AppBar>
  </div>
);

OverviewFooterComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  appBar: {
    backgroundColor: '#A8C6FA', // TODO: move color to theme
  },
  button: {
    margin: theme.spacing.unit,
    width: '100%',
  },
}))(OverviewFooterComponent);
