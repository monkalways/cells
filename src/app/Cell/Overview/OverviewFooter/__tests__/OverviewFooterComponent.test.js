import React from 'react';
import { shallow } from 'enzyme';

import { BottomNavigationAction } from '@material-ui/core';
import { OverviewFooterComponent } from '../OverviewFooterComponent';

describe('OverviewFooterComponent', () => {
  let isAuthenticated;
  let cellName;
  let onSignIn;
  let push;

  beforeEach(() => {
    isAuthenticated = true;
    cellName = 'c1';
    onSignIn = jest.fn();
    push = jest.fn();
  });

  const setup = () => {
    const classes = {
      root: '',
      appBar: '',
      navigation: '',
      icon: '',
    };
    return shallow(<OverviewFooterComponent
      classes={classes}
      history={{ push }}
      onSignIn={onSignIn}
      cellName={cellName}
      isAuthenticated={isAuthenticated}
    />);
  };

  it('should render sign in button if not authenticated', () => {
    isAuthenticated = false;
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('#signInButton')).toExist();
    expect(wrapper.find(BottomNavigationAction)).not.toExist();
  });

  it('should handle sign in if not authenticated', () => {
    isAuthenticated = false;
    const wrapper = setup();

    wrapper.find('#signInButton').simulate('click');

    expect(onSignIn).toBeCalled();
  });

  it('should render bottom navigation actions if authenticated', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('#cellCheckButton')).toExist();
    expect(wrapper.find('#mealButton')).toExist();
    expect(wrapper.find('#medicationButton')).toExist();

    expect(wrapper.find('#signInButton')).not.toExist();
  });

  it('should handleCellCheckClick if authenticated', () => {
    const wrapper = setup();

    wrapper.find('#cellCheckButton').simulate('click');
    expect(push).toBeCalledWith('/cells/c1/home/cell-check');
  });

  it('should handleMealClick if authenticated', () => {
    const wrapper = setup();

    wrapper.find('#mealButton').simulate('click');
    expect(push).toBeCalledWith('/cells/c1/home/meal');
  });

  it('should handleMedicationClick if authenticated', () => {
    const wrapper = setup();

    wrapper.find('#medicationButton').simulate('click');
    expect(push).toBeCalledWith('/cells/c1/home/medication');
  });
});
