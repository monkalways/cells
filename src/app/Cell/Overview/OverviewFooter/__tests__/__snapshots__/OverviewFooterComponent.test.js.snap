// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewFooterComponent should render bottom navigation actions if authenticated 1`] = `
<div
  className="root"
>
  <WithStyles(AppBar)
    className="appBar"
    position="static"
  >
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      justify="center"
    >
      <WithStyles(BottomNavigation)
        className="navigation"
        showLabels={true}
        value={-1}
      >
        <WithStyles(BottomNavigationAction)
          icon={
            <img
              alt="wellness visual"
              className="navigationImage"
              src="WellnessVisual.png"
            />
          }
          id="cellCheckButton"
          label={
            <WithStyles(Typography)
              variant="body1"
            >
              Cell Check
            </WithStyles(Typography)>
          }
          onClick={[Function]}
        />
        <WithStyles(BottomNavigationAction)
          icon={
            <img
              alt="meal accept"
              className="navigationImage"
              src="MealAccept.png"
            />
          }
          id="mealButton"
          label={
            <WithStyles(Typography)
              variant="body1"
            >
              Meal
            </WithStyles(Typography)>
          }
          onClick={[Function]}
        />
        <WithStyles(BottomNavigationAction)
          className="disabled"
          disabled={true}
          icon={
            <img
              alt="Medication accept"
              className="navigationImage"
              src="MedicineAccept.png"
            />
          }
          id="medicationButton"
          label={
            <WithStyles(Typography)
              variant="body1"
            >
              Medication
            </WithStyles(Typography)>
          }
          onClick={[Function]}
        />
      </WithStyles(BottomNavigation)>
    </WithStyles(Grid)>
  </WithStyles(AppBar)>
</div>
`;

exports[`OverviewFooterComponent should render sign in button if not authenticated 1`] = `
<div
  className="root"
>
  <WithStyles(AppBar)
    className="appBar"
    position="static"
  >
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      justify="center"
    >
      <WithStyles(Button)
        className="button"
        color="secondary"
        id="signInButton"
        onClick={[Function]}
        variant="contained"
      >
        Scan ID Card to Access
      </WithStyles(Button)>
      <Connect(WithStyles(ScanCardDialogComponent)) />
    </WithStyles(Grid)>
  </WithStyles(AppBar)>
</div>
`;
